src/gpu/generic/sycl/ref_convolution.hpp

```diff
@@ -0,0 +1,383 @@
+    using gpu::generic::sycl::primitive_t::primitive_t;
+
+    struct pd_t : public convolution_bwd_data_pd_t {
+        using convolution_bwd_data_pd_t::convolution_bwd_data_pd_t;
+
+        DECLARE_COMMON_PD_T("dpcpp:ref:any", ref_convolution_bwd_data_t);
+
+        status_t init(impl::engine_t *engine) {
+            using namespace data_type;
+            using sm = primitive_attr_t::skip_mask_t;
+
+            const memory_desc_wrapper diff_data_d(diff_src_md());
+            const memory_desc_wrapper weights_d(weights_md());
+            const memory_desc_wrapper diff_dst_d(diff_dst_md());
+
+            const bool ok = is_bwd_d() && check_work_amount(weights_d)
+                    && set_default_formats()
+                    && check_data_types(diff_data_d, weights_d, diff_dst_d)
+                    && check_formats(diff_data_d, weights_d, diff_dst_d)
+                    && attr()->has_default_values(sm::scales_runtime
+                            | sm::zero_points_runtime | sm::post_ops
+                            | sm::sum_dt)
+                    && IMPLICATION(!attr()->scales_.has_default_values(),
+                            check_scales_mask())
+                    && post_ops_ok();
+            if (!ok) return status::unimplemented;
+
+            return init_conf();
+        }
+
+        sycl_convolution_conf_t conf_;
+
+    private:
+        status_t init_conf();
+
+        bool set_default_formats() {
+            using namespace format_tag;
+            auto dat_tag = utils::pick(ndims() - 3, nwc, nhwc, ndhwc);
+            auto wei_tag = with_groups()
+                    ? utils::pick(ndims() - 3, goiw, goihw, goidhw)
+                    : utils::pick(ndims() - 3, oiw, oihw, oidhw);
+            return set_default_formats_common(dat_tag, wei_tag, dat_tag);
+        }
+
+        bool check_scales_mask() const {
+            const std::vector<int> supported_args
+                    = {DNNL_ARG_SRC_0, DNNL_ARG_WEIGHTS, DNNL_ARG_DST};
+            return attr_scales_ok(supported_args);
+        }
+
+        bool post_ops_ok() const {
+            for (int i = 0; i < attr()->post_ops_.len(); i++) {
+                const auto &e = attr()->post_ops_.entry_[i];
+                if (!IMPLICATION(e.is_eltwise(),
+                            utils::one_of(e.eltwise.alg, alg_kind::eltwise_relu,
+                                    alg_kind::eltwise_linear,
+                                    alg_kind::eltwise_clip,
+                                    alg_kind::eltwise_clip_v2,
+                                    alg_kind::eltwise_hardswish))) {
+                    return false;
+                }
+            }
+            return attr()->post_ops_.len() <= sycl_post_ops_t::max_post_ops
+                    && attr()->post_ops_.has_default_values(
+                            {primitive_kind::eltwise, primitive_kind::prelu,
+                                    primitive_kind::sum});
+        }
```

Comment:
instead of adding checks for what is supported by sycl_post_ops_t in all implementations using it, could you add `sycl_post_ops_t::post_ops_ok(post_ops_t) ` function that will be called by implementations relying on sycl_post_ops_t?

This will make it more scalable as support for more eltwise/binary is added  to sycl_post_ops.

Reply:
I want to make that change, but it makes more sense to do it for all primitives at once in a separate PR.